rls/src/main/java/io/grpc/rls/RlsProtoConverters.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.rls;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.base.Converter;
+import io.grpc.internal.JsonUtil;
+import io.grpc.lookup.v1.RouteLookupRequest;
+import io.grpc.lookup.v1.RouteLookupResponse;
+import io.grpc.rls.RlsProtoData.GrpcKeyBuilder;
+import io.grpc.rls.RlsProtoData.GrpcKeyBuilder.Name;
+import io.grpc.rls.RlsProtoData.NameMatcher;
+import io.grpc.rls.RlsProtoData.RequestProcessingStrategy;
+import io.grpc.rls.RlsProtoData.RouteLookupConfig;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A RlsProtoConverters is a collection of {@link Converter} between RouteLookupService proto
+ * messages to internal representation in {@link RlsProtoData}.
+ */
+public final class RlsProtoConverters {
```

Comment:
I suggest to move most classes in this package to `io.grpc.rls.internal.*`, as #6711, it could be very difficult to do it later due to pending merge conflicts.

Reply:
good point. thanks removed to internal.